drivers/mbox/mbox_nxp_fsl_mu.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Wrapper of the Freescale Message Unit driver into Zephyr's MBOX model.
+ */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/mbox.h>
+#include <zephyr/sys/util_macro.h>
+#include <fsl_mu.h>
+
+#define LOG_LEVEL CONFIG_MBOX_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(nxp_fsl_mu);
+
+#define DT_DRV_COMPAT nxp_fsl_mu
+
+#define MU_MAX_CHANNELS 2
+
+#define MU_MAX_MBOX_PER_CHAN	1
+#define MU_MBOX_SIZE		4
+
+#define MU_NODE(n)             DT_NODELABEL(mu##n)
+#define MU_BASE(n)             ((MU_Type *)DT_REG_ADDR(MU_NODE(n)))
+
+struct nxp_fsl_mu_data {
+	mbox_callback_t cb[MU_MAX_CHANNELS];
+	void *user_data[MU_MAX_CHANNELS];
+};
+
+struct nxp_fsl_mu_config {
+	MU_Type *base;
+	void (*config_irq)(void);
+};
+
+static int nxp_fsl_mu_send(const struct device *dev, uint32_t channel,
+			    const struct mbox_msg *msg)
+{
+	const struct nxp_fsl_mu_config *cfg = dev->config;
+	uint32_t data;
+
+	if (channel >= MU_MAX_CHANNELS) {
+		return -EINVAL;
+	}
+
+	/* Signalling mode. */
+	if (msg == NULL) {
+		if (kStatus_Success != MU_TriggerInterrupts(
```

Comment:
`MU_TriggerInterrupts()` already checks if the previous interrupt has been processed - https://github.com/nxp-mcuxpresso/mcux-sdk/blob/main/drivers/mu/fsl_mu.c#L167

Reply:
I understand that `MU_TriggerInterrupts()` will check and return an error if the previous interrupt has not been processed. I wanted to notify the caller of mbox_send() if that's the case. Would you like me to just do
`return MU_TriggerInterrupts(cfg->base, kMU_GenInt0InterruptTrigger);` ? That will cast the status_t to an int.

Reply:
Yes, I would call it directly, no need for an extra check